codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -0,0 +1,642 @@
+        val fnName = "serialize_${operationShape.id.name.toSnakeCase()}_error"
+        val errorSymbol = operationShape.errorSymbol(symbolProvider)
+        return RuntimeType.forInlineFun(fnName, operationSerModule) {
+            Attribute.Custom("allow(clippy::unnecessary_wraps)").render(it)
+            it.rustBlockTemplate(
+                "pub fn $fnName(error: &#{E}Kind) -> std::result::Result<#{Http}::Response<#{Bytes}>, #{Error}>",
+                *codegenScope,
+                "E" to errorSymbol
+            ) {
+                serverRenderShapeErrorSerializer(
+                    operationShape,
+                    errorSymbol,
+                )
+            }
+        }
+    }
+
+    private fun RustWriter.serverRenderShapeErrorSerializer(
+        operationShape: OperationShape,
+        errorSymbol: RuntimeType,
+    ) {
+        val structuredDataSerializer = protocol.structuredDataSerializer(operationShape)
+        rustTemplate("let mut response = #{Http}::Response::builder();", *codegenScope)
+        rust("let mut out = String::new();")
+        rustTemplate("let mut object = #{JsonObjectWriter}::new(&mut out);", *codegenScope)
+        withBlock("match error {", "};") {
+            operationShape.errors.forEach {
+                val variantShape = model.expectShape(it, StructureShape::class.java)
+                val errorTrait = variantShape.expectTrait<ErrorTrait>()
+                val variantSymbol = symbolProvider.toSymbol(variantShape)
+                val data = safeName("var")
+                val serializerSymbol = structuredDataSerializer.serverErrorSerializer(it)
+                rustBlock("#TKind::${variantSymbol.name}($data) =>", errorSymbol) {
+                    rust(
+                        """
+                            #T(&$data)?;
+                            object.key(${"code".dq()}).string(${httpBindingResolver.errorCode(variantShape).dq()});
+                        """.trimIndent(),
+                        serializerSymbol
+                    )
+                    if (variantShape.errorMessageMember() != null) {
+                        rust(
+                            """
+                                if let Some(message) = $data.message() {
+                                    object.key(${"message".dq()}).string(message);
+                                }
+                            """.trimIndent()
+                        )
+                    }
+                    val bindings = httpBindingResolver.errorResponseBindings(it)
+                    bindings.forEach { binding ->
+                        when (val location = binding.location) {
+                            HttpLocation.RESPONSE_CODE, HttpLocation.DOCUMENT -> {}
+                            else -> {
+                                logger.warning(
+                                    "$operationShape: response serialization does not currently support $location bindings"
+                                )
```

Comment:
rather than generating nothing here, I'd generate a comment or a `todo!()`. Makes it easier to debug the generated code